File: ProductAngular/ProjectAPI/ProjectAPI/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace ProjectAPI.Models
{
    public class Product
    {

        [Key]
        public int ProductId { get; set; }
        
        public string ProductName { get; set; }
       
        public DateTime PurchaseDate { get; set; }
        public bool ProductAvailable { get; set; }
    
        public int SupplierId { get; set; }
        [ForeignKey("SupplierId")]
        public string ContactName { get; set; }
        
        public int Quantity { get; set; }
        
        public decimal UnitPrice { get; set; }
       
        public decimal SalesUnitPrice { get; set; }
        public string Photos { get; set; }
        public virtual Supplier Supplier { get; set; }
    }
}

File: ProductAngular/src/app/model/product.ts
export class product{
    ProductId?:number;
    ProductName?:string;
    PurchaseDate?:Date;
    ProductAvailable?:boolean;
    SupplierId?:number;
    ContactName?:string;
    Quantity?: number;
    UnitPrice?: number;
    SalesUnitPrice?: number;
    Photos?:string;


    


    constructor(
    _ProductId?:number,
    _ProductName?:string,
    _PurchaseDate?:Date,
    _ProductAvailable?:boolean,
    _SupplierId?:number,
    _ContactName?:string,
    _Quantity?: number,
    _UnitPrice?: number,
    _SalesUnitPrice?: number,
    _Photos?:string,
        
    )
    {
       this.ProductId=_ProductId;
       this.ProductName=_ProductName;
       this.PurchaseDate=_PurchaseDate;
       this.ProductAvailable=_ProductAvailable;
       this.SupplierId=_SupplierId;
       this.ContactName=_ContactName;
       this.Quantity=_Quantity;
       this.UnitPrice=_UnitPrice;
       this.SalesUnitPrice=_SalesUnitPrice;
       this.Photos=_Photos;
       
    }



}
